librz/core/cmd_type.c

```diff
@@ -2081,25 +1215,80 @@ RZ_IPI RzCmdStatus rz_type_noreturn_del_all_handler(RzCore *core, int argc, cons
 }
 
 RZ_IPI RzCmdStatus rz_type_open_file_handler(RzCore *core, int argc, const char **argv) {
-	types_open_file(core, argv[1]);
+	rz_types_open_file(core, argv[1]);
 	return RZ_CMD_STATUS_OK;
 }
 
 RZ_IPI RzCmdStatus rz_type_open_editor_handler(RzCore *core, int argc, const char **argv) {
 	const char *typename = argc > 1 ? argv[1] : NULL;
-	types_open_editor(core, typename);
+	rz_types_open_editor(core, typename);
 	return RZ_CMD_STATUS_OK;
 }
 
 RZ_IPI RzCmdStatus rz_type_open_sdb_handler(RzCore *core, int argc, const char **argv) {
-	types_open_sdb(core, argv[1]);
+	rz_types_open_sdb(core, argv[1]);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_print_handler(RzCore *core, int argc, const char **argv) {
+	const char *addr_or_var = argc > 2 ? argv[2] : NULL;
+	if (!addr_or_var) {
+		return type_format_print(core, argv[1], core->offset);
+	}
+	ut64 addr = rz_num_math(core->num, addr_or_var);
+	if (!addr) {
+		return type_format_print_variable(core, argv[1], addr_or_var);
+	}
+	return type_format_print(core, argv[1], addr);
```

Comment:
no test for this line, might be worth adding a simple one given the very big refactoring.